File: java/Kode/Sword.java
public class Sword extends Weapon {
    private double sharpness;
    private double attackSpeed;
    private double blockChance;

    public Sword(String name, int level, String tier, WeaponMaterial material, 
                 double sharpness, double attackSpeed, double blockChance) {
        super(name, level, tier, material);
        this.sharpness = sharpness;
        this.attackSpeed = attackSpeed;
        this.blockChance = blockChance;
    }

    @Override
    protected String specificsInfo() {
        return "Sharpness: " + sharpness + ", Attack Speed: " + attackSpeed + ", Block Chance: " + blockChance;
    }

    // Getter untuk kebutuhan Hybrid
    public double getSharpness() { return sharpness; }
    public double getAttackSpeed() { return attackSpeed; }
    public double getBlockChance() { return blockChance; }
}

File: cpp/Kode/Sword.cpp
#include "Sword.h"
#include "WeaponMaterial.h" 

Sword::Sword(std::string n, int l, std::string t, const std::vector<std::pair<std::string, double>>& s, 
      std::unique_ptr<WeaponMaterial> m,
      double sh, double as, double bc) : 
    Weapon(n, l, t, s, std::move(m)), sharpness(sh), AttackSpeed(as), BlockChance(bc) {}

void Sword::displayWeapon() const {
    Weapon::displayWeapon(); 
    std::cout << "  Specifics: Sharpness: " << sharpness << ", Attack Speed: " << AttackSpeed << ", Block Chance: " << BlockChance << std::endl;
}

void Sword::displayWeaponWithoutMaterial() const {
    Weapon::displayWeaponWithoutMaterial(); 
    std::cout << "  Specifics: Sharpness: " << sharpness << ", Attack Speed: " << AttackSpeed << ", Block Chance: " << BlockChance << std::endl;
}